### PR TITLE
Properly propagate more failures and unit test all the things!

### PR DESCRIPTION
**Pull request by petr-muller**
Previously, when we failed to submit some rehearsal jobs, that failure could be covered up later by all successfully submitted rehearsals passing.

Additionally, added a bunch of unit tests for `ExecuteJobs` to do more testing of various, mostly negative, cases.

/cc @droslean @stevekuznetsov @bbguimaraes

### CODE REVIEW

File: pkg/rehearse/jobs_test.go [hunk 1]
```diff
@@ -254,38 +255,175 @@ func makeTestingProwJob(name, namespace, jobName, context string, refs *pjapi.Re
...
+		t.Run(tc.description, func(t *testing.T) {
+			testLogger := logrus.New()
+			fakecs := fake.NewSimpleClientset()
+			fakeclient := fakecs.ProwV1().ProwJobs(testNamespace)
+			watcher, err := fakeclient.Watch(metav1.ListOptions{})
+			if err != nil {
+				t.Fatalf("Failed to setup watch: %v", err)
+			}
+			fakecs.Fake.PrependWatchReactor("prowjobs", makeSuccessfulFinishReactor(watcher, tc.jobs))
+			fakecs.PrependReactor("create", "prowjobs", tc.reactor)
```
Comment (stevekuznetsov) [inline pkg/rehearse/jobs_test.go]: Could explicitly call out to `Fake.PrependReactor` or not above, to be consistent

File: pkg/rehearse/jobs_test.go [hunk 2]
```diff
@@ -254,38 +255,175 @@ func makeTestingProwJob(name, namespace, jobName, context string, refs *pjapi.Re
...
+}
+
+func TestExecuteJobsErrors(t *testing.T) {
+	testPrNumber, testNamespace, testRepoPath, testRefs := makeTestData()
+	targetRepo := "targetOrg/targetRepo"
+
+	testCases := []struct {
+		description string
+		jobs        map[string][]prowconfig.Presubmit
+		reactor     func(action clientgo_testing.Action) (handled bool, ret runtime.Object, err error)
```
Comment (stevekuznetsov) [inline pkg/rehearse/jobs_test.go]: You might be able to DRY this out to a list of job names that could fail on create and make the test cases more narrow

File: pkg/rehearse/jobs_test.go [hunk 3]
```diff
@@ -325,67 +462,58 @@ func TestExecuteJobs(t *testing.T) {
...
+				return
+			}
 
-				// Overwrite dynamic struct members to allow comparison
-				for i := range createdJobs.Items {
-					createdJobs.Items[i].Name = generatedName
-					createdJobs.Items[i].Status.StartTime.Reset()
-				}
+			// Overwrite dynamic struct members to allow comparison
+			for i := range createdJobs.Items {
```
Comment (stevekuznetsov) [inline pkg/rehearse/jobs_test.go]: You could also just validate `[]prowapi.ProwJobSpec`

**Review by stevekuznetsov — COMMENTED** [review]
Nice1

File: pkg/rehearse/jobs_test.go (hunk 3 above)
Comment (petr-muller) [reply pkg/rehearse/jobs_test.go]: Yeah, that simplifies everything a bit! :+1:

**Comment by petr-muller** [comment]
@stevekuznetsov Addressed the feedback, thanks. Given some of the simplifications touch tests not related to this PR, I prefer to keep the simplifications in a separate commit, not squash it.

File: pkg/rehearse/jobs_test.go [hunk 4]
```diff
@@ -330,7 +320,17 @@ func TestExecuteJobsErrors(t *testing.T) {
 				t.Fatalf("Failed to setup watch: %v", err)
 			}
 			fakecs.Fake.PrependWatchReactor("prowjobs", makeSuccessfulFinishReactor(watcher, tc.jobs))
-			fakecs.PrependReactor("create", "prowjobs", tc.reactor)
+			fakecs.Fake.PrependReactor("create", "prowjobs", func(action clientgo_testing.Action) (bool, runtime.Object, error) {
+				createAction := action.(clientgo_testing.CreateAction)
+				pj := createAction.GetObject().(*pjapi.ProwJob)
+				for _, jobName := range tc.failToCreate {
```
Comment (stevekuznetsov) [inline pkg/rehearse/jobs_test.go]: If you use `sets.String` this becomes `tc.failToCreate.Has(pj.Spec.Job)`

**Review by stevekuznetsov — COMMENTED** [review]
Minor nit. otherwise LGTM

File: pkg/rehearse/jobs_test.go (hunk 4 above)
Comment (petr-muller) [reply pkg/rehearse/jobs_test.go]: Sweet. It'll take a while until I memorize all the kube apis.
